__init__.py

```diff
@@ -156,143 +137,131 @@ def cancel_override(self):
-                                      self.handle_device_homescreen_settings)
-            self.gui.register_handler('mycroft.device.settings.ssh',
-                                      self.handle_device_ssh_settings)
-            self.gui.register_handler('mycroft.device.settings.reset',
-                                      self.handle_device_factory_reset_settings)
-            self.gui.register_handler('mycroft.device.settings.update',
-                                      self.handle_device_update_settings)
-            self.gui.register_handler('mycroft.device.settings.restart',
-                                      self.handle_device_restart_action)
-            self.gui.register_handler('mycroft.device.settings.poweroff',
-                                      self.handle_device_poweroff_action)
-            self.gui.register_handler('mycroft.device.settings.wireless',
-                                      self.handle_show_wifi_screen_intent)
-            self.gui.register_handler('mycroft.device.show.idle',
-                                      self.resting_screen.show)
+            self.gui.register_handler(
+                "mycroft.device.settings", self.handle_device_settings
+            )
+            self.gui.register_handler(
+                "mycroft.device.settings.homescreen",
+                self.handle_device_homescreen_settings,
+            )
+            self.gui.register_handler(
+                "mycroft.device.settings.ssh", self.handle_device_ssh_settings
+            )
+            self.gui.register_handler(
+                "mycroft.device.settings.reset",
+                self.handle_device_factory_reset_settings,
+            )
+            self.gui.register_handler(
+                "mycroft.device.settings.update", self.handle_device_update_settings
+            )
+            self.gui.register_handler(
+                "mycroft.device.settings.restart", self.handle_device_restart_action
+            )
+            self.gui.register_handler(
+                "mycroft.device.settings.poweroff", self.handle_device_poweroff_action
+            )
+            self.gui.register_handler(
+                "mycroft.device.settings.wireless", self.handle_show_wifi_screen_intent
+            )
+            self.gui.register_handler(
+                "mycroft.device.show.idle", self.resting_screen.show
+            )
 
             # Handle idle selection
-            self.gui.register_handler('mycroft.device.set.idle',
-                                      self.resting_screen.set)
+            self.gui.register_handler(
+                "mycroft.device.set.idle", self.resting_screen.set
+            )
 
             # System events
-            self.add_event('system.reboot', self.handle_system_reboot)
-            self.add_event('system.shutdown', self.handle_system_shutdown)
-
-            # Handle volume setting via I2C
-            self.add_event('mycroft.volume.set', self.on_volume_set)
```

Comment:
I get it but it is not clear to me this is not the cause of the message bus tasks i am now seeing on core exit. I have 2 questions. First if someone publishes messages nobody consumes what becomes of them? Do they grow the internal q until it blows? and second, if I publish a message with an expected response (I believe I have seen this design pattern) and the response never comes what becomes of the response handler? I don't know enough about pyee to figure it out but it looks like messages need to be consumed eventually.

Reply:
Messages don't have an expected recipient, they just "are". 
So I can publish any number of messages on the bus, whether something reacts to that message or not is completely up to them and doesn't affect the message or the message bus.

The expected response method (`wait_for_response()`) has a timeout, defaults to 3 seconds. So will return None if it exceeds the timeout.

Event handlers are open ended, they remain until they are removed or the system is shutdown.